crates/ra_lsp_server/tests/heavy_tests/main.rs

```diff
@@ -266,17 +266,18 @@ fn test_missing_module_code_action_in_json_project() {
     let code = format!(
         r#"
 //- rust-project.json
-{{ 
-    "roots": [ "{PATH}" ], 
-    "crates": [ {{ "root_module": "{PATH}/src/lib.rs", "deps": [], "edition": "2015" }} ] 
+{{
+    "roots": [ "{PATH}" ],
+    "crates": [ {{ "root_module": "{PATH}/src/lib.rs", "deps": [], "edition": "2015" }} ]
 }}
 
 //- src/lib.rs
 mod bar;
 
-fn main() {}
+fn main() {{}}
 "#,
-        PATH = tmp_dir.path().display()
+        // NOTE: Escape windows paths properly for JSON
+        PATH = tmp_dir.path().to_string_lossy().replace("\\", "\\\\")
```

Comment:
Do we have `json!` macro already among depenencies? Could we use it here?

Reply:
Yeah we do have the macro, unfortunately it seems difficulty to get the JSON string without quotes but with the escaped path.

```rust
json!(tmp_dir.path()) 
// -> returns "C:\\Users\\v\\AppData\\Local\\Temp\\.tmpzeVlJC"
// Works fine for "roots" but not "root_module" which turns into
// "C:\\Users\\v\\AppData\\Local\\Temp\\.tmpzeVlJC"/src/lib.rs
```
and when using 
```rust
json!(tmp_dir.path()).as_str().unwrap()
```
it has the same effect as `path().display()`

Reply:
I mean, shound't 

```
json! {{
 "roots": [path().display()]
}}
```

just work?

Reply:
An alternative is to introduce another format parameter for the `root_module` and then using debug formatter, which should escape the string. 

e.g. 

```rust
    let code = format!(
        r#"
//- rust-project.json
{{
    "roots": [ {PATH:?} ],
    "crates": [ {{ "root_module": {ROOT:?}, "deps": [], "edition": "2015" }} ]
}}

//- src/lib.rs
mod bar;

fn main() {{}}
"#,
        PATH = tmp_dir.path().display(),
        ROOT = format!("{}/src/lib.rs", tmp_dir.path().display())
    );
```

Reply:
> I mean, shound't
> 
> ```
> json! {{
>  "roots": [path().display()]
> }}
> ```
> just work?

Ah, let's see!

Reply:
hm, `json!` seems to work as expected: use serde_json::json;

fn main() {
    // The type of `john` is `serde_json::Value`
    let john = json!({
        "name": "John Doe",
        "age": 43,
        "phones": [
            "+44 1234567",
            "+44 2345678"
        ]
    });

    println!("first phone number: {}", john["phones"][0]);

    // Convert to a string of JSON and print it out
    println!("{}", john.to_string());
}